### PR TITLE
adding blobtk plotting functionality to the blobtoolkit

### PR DESCRIPTION
**Pull request by SaimMomin12**
This PR relates to adding of `blobtk plot` tool functionality to the existing Blobtoolkit tool.

### CODE REVIEW

File: tools/blobtoolkit/macros.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
     
     <xml name="requirements">
         <requirements>
-            <container type="docker">genomehubs/blobtoolkit:@TOOL_VERSION@</container>
+            <container type="docker">quay.io/biocontainers/blobtoolkit:4.3.9</container>
```
Comment (bgruening) [inline tools/blobtoolkit/macros.xml:7]: use the 11 one

```suggestion
            <container type="docker">quay.io/biocontainers/blobtoolkit:4.3.11</container>
```

File: tools/blobtoolkit/blobtoolkit.xml [hunk 2]
```diff
@@ -165,6 +170,7 @@
                 <option value="create">Create a BlobToolKit dataset</option>
                 <option value="add">Add data to a BlobToolKit dataset</option>
                 <option value="filter">Edit a BlobToolKit dataset</option>
+                <option value="plot">Plot a BlobToolKit dataset</option>
```
Comment (bgruening) [inline tools/blobtoolkit/blobtoolkit.xml:173]: should the plot not be added __in addition__? I mean create/add/edit a dataset and in addition plot it ...?

**Comment by SaimMomin12** [comment]
Sorry I realise I missed commit some files, I have marked this PR as draft for now

File: tools/blobtoolkit/macros.xml (hunk 1 above)
Comment (SaimMomin12) [reply tools/blobtoolkit/macros.xml:7]: Okay

**Comment by SaimMomin12** [comment]
I used the latest container v4.3.11, but it still throws the following error only for blobtk 

```
Job in error state.. tool_id: blobtoolkit, exit_code: 134, stderr: thread 'main' panicked at 'unable to parse json: Error("missing field `accession`", line: 7, column: 2)', src/blobdir.rs:180:58
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Aborted (core dumped)
.
```

Additional, when I checked the version of blobtk that in the container, it shows version as `0.2.4` but we intent to use the latest version `0.5.3`

I assume, the error above is perhaps due to the older version of blobtk. Can we update the version of blobtk in the container to the latest version?

File: tools/blobtoolkit/blobtoolkit.xml (hunk 2 above)
Comment (SaimMomin12) [reply tools/blobtoolkit/blobtoolkit.xml:174]: I think it would be better to keep the plotting option as a seperate functionality

Comment (bgruening) [reply tools/blobtoolkit/blobtoolkit.xml:174]: And what about both? Adding it as separate functionality AND as option after the other function?

Comment (SaimMomin12) [reply tools/blobtoolkit/blobtoolkit.xml:174]: Yes I agree! We can ask if the user wants to create/edit/filter **AND** plot the dataset simultaneously. If yes, we can add blobtk to the command and if not then we simply output the dataset. How does this sound?

**Review by bgruening — APPROVED** [review]

File: tools/blobtoolkit/blobtoolkit.xml (hunk 2 above)
Comment (bgruening) [reply tools/blobtoolkit/blobtoolkit.xml:174]: Your call, do you want to add it? The changes look good.

Comment (SaimMomin12) [reply tools/blobtoolkit/blobtoolkit.xml:187]: I have added the plotting functionality to add/filter options. Now the user has option to plot the blobdataset along with add/filter functionality as well as standalone plot the blobdataset.

File: tools/blobtoolkit/blobtoolkit.xml [hunk 3]
```diff
@@ -22,7 +22,15 @@
             --taxid $mode_conditional.taxid
             --taxdump './taxdump'
             './Blobdir'
+            
             && tar -C ./Blobdir -zcvf Blobdir.tgz .
+
+        #else if $mode_conditional.selector == 'plot'
+            blobtk --version &&
```
Comment (bgruening) [inline tools/blobtoolkit/blobtoolkit.xml:29]:
```suggestion
```

File: tools/blobtoolkit/blobtoolkit.xml [hunk 4]
```diff
@@ -22,7 +22,15 @@
...
+            mkdir -p blobtk_files && 
+            tar -xvf "$blobdir_input" -C blobtk_files &&
```
Comment (bgruening) [inline tools/blobtoolkit/blobtoolkit.xml:31]:
```suggestion
            tar -xvf '$blobdir_input' -C blobtk_files &&
```

File: tools/blobtoolkit/blobtoolkit.xml [hunk 5]
```diff
@@ -22,7 +22,15 @@
...
+            blobtk plot --blobdir blobtk_files  --view "$plot_view"
```
Comment (bgruening) [inline tools/blobtoolkit/blobtoolkit.xml:32]:
```suggestion
            blobtk plot --blobdir blobtk_files  --view '$plot_view'
```

File: tools/blobtoolkit/blobtoolkit.xml [hunk 6]
```diff
@@ -156,15 +169,22 @@
                 #end if
                 ./Blobdir
             #end if
+
+            #if $mode_conditional.blobtk_plot_options.blobtk_plot == 'yes'
+                && blobtk plot --blobdir './Blobdir' --view $blobtk_plot_options.plot_view
```
Comment (bgruening) [inline tools/blobtoolkit/blobtoolkit.xml:174]:
```suggestion
                && blobtk plot --blobdir ./Blobdir --view $blobtk_plot_options.plot_view
```

**Review by bgruening — APPROVED** [review]

**Pull request merged**